File: app/src/main/java/com/jann_luellmann/thekenapp/util/Prefs.java
package com.jann_luellmann.thekenapp.util;

import android.content.Context;
import android.content.SharedPreferences;

public class Prefs
{
    public static final String CURRENT_EVENT = "CURRENT_EVENT";

    private static SharedPreferences getPreferences(Context context)
    {
        return context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    public static void putString(Context context, String key, String value)
    {
        getPreferences(context).edit().putString(key, value).apply();
    }

    public static void putInt(Context context, String key, int value)
    {
        getPreferences(context).edit().putInt(key, value).apply();
    }

    public static void putLong(Context context, String key, long value)
    {
        getPreferences(context).edit().putLong(key, value).apply();
    }

    public static boolean containsKey(Context context, String key)
    {
        try
        {
            return getPreferences(context).contains(key);
        }
        catch(Exception e)
        {
            return false;
        }
    }

    public static String getString(Context context, String key)
    {
        return getPreferences(context).getString(key, null);
    }

    public static String getString(Context context, String key, String defaultValue)
    {
        return getPreferences(context).getString(key, defaultValue);
    }

    public static Boolean getBoolean(Context context, String key, Boolean defaultValue)
    {
        return getPreferences(context).getBoolean(key, defaultValue);
    }

    public static int getInt(Context context, String key, int defaultValue)
    {
        return getPreferences(context).getInt(key, defaultValue);
    }

    public static long getLong(Context context, String key, long defaultValue)
    {
        return getPreferences(context).getLong(key, defaultValue);
    }

    public static void putBoolean(Context context, String key, Boolean value)
    {
        getPreferences(context).edit().putBoolean(key, value).apply();
    }

    public static float getDouble(Context context, String key, float defaultValue)
    {
        return getPreferences(context).getFloat(key, defaultValue);
    }

    public static void putDouble(Context context, String key, float value)
    {
        getPreferences(context).edit().putFloat(key, value).apply();
    }

    public static void deleteKey(Context context, String key)
    {
        getPreferences(context).edit().remove(key).apply();
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/util/Prefs.kt
package com.jann_luellmann.thekenapp.util

import android.content.Context
import android.content.SharedPreferences

object Prefs {
    const val CURRENT_EVENT = "CURRENT_EVENT"

    private fun getPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences(context.packageName, Context.MODE_PRIVATE)
    }

    fun putCurrentEvent(context: Context, eventId: Long) {
        getPreferences(context).edit().putLong(CURRENT_EVENT, eventId).apply()
    }

    fun getCurrentEvent(context: Context): Long {
        return getPreferences(context).getLong(CURRENT_EVENT, -1)
    }

    /** ---------------------------------------------------------------------- **/

//    fun putString(context: Context, key: String?, value: String?) {
//        getPreferences(context).edit().putString(key, value).apply()
//    }
//
//    fun putInt(context: Context, key: String?, value: Int) {
//        getPreferences(context).edit().putInt(key, value).apply()
//    }
//
//    fun putLong(context: Context, key: String?, value: Long) {
//        getPreferences(context).edit().putLong(key, value).apply()
//    }
//
//    fun containsKey(context: Context, key: String?): Boolean {
//        return try {
//            getPreferences(context).contains(key)
//        } catch (e: Exception) {
//            false
//        }
//    }
//
//    fun getString(context: Context, key: String?): String? {
//        return getPreferences(context).getString(key, null)
//    }
//
//    fun getString(context: Context, key: String?, defaultValue: String?): String? {
//        return getPreferences(context).getString(key, defaultValue)
//    }
//
//    fun getBoolean(context: Context, key: String?, defaultValue: Boolean?): Boolean {
//        return getPreferences(context).getBoolean(key, defaultValue!!)
//    }
//
//    fun getInt(context: Context, key: String?, defaultValue: Int): Int {
//        return getPreferences(context).getInt(key, defaultValue)
//    }
//
//    fun getLong(context: Context, key: String?, defaultValue: Long): Long {
//        return getPreferences(context).getLong(key, defaultValue)
//    }
//
//    fun putBoolean(context: Context, key: String?, value: Boolean?) {
//        getPreferences(context).edit().putBoolean(key, value!!).apply()
//    }
//
//    fun getDouble(context: Context, key: String?, defaultValue: Float): Float {
//        return getPreferences(context).getFloat(key, defaultValue)
//    }
//
//    fun putDouble(context: Context, key: String?, value: Float) {
//        getPreferences(context).edit().putFloat(key, value).apply()
//    }
//
//    fun deleteKey(context: Context, key: String?) {
//        getPreferences(context).edit().remove(key).apply()
//    }
}